src/vm/slot.rs

```diff
@@ -11,44 +11,60 @@ use atomic::Atomic;
 use crate::util::constants::{BYTES_IN_ADDRESS, LOG_BYTES_IN_ADDRESS};
 use crate::util::{Address, ObjectReference};
 
-/// A `Slot` represents a slot in an object (a.k.a. a field), on the stack (i.e. a local variable)
-/// or any other places (such as global variables).  A slot may hold an object reference. We can
-/// load the object reference from it, and we can update the object reference in it after the GC
-/// moves the object.
+/// A `Slot` value *points to* a slot, and is intended for loading and updating the reference held
+/// in the slot.  Note that a `Slot` is not the slot itself.  It is a stateless data type that
+/// usually contains a pointer to the actual slot.  It can be [copied](std::marker::Copy), and the
+/// copied `Slot` instance points to the same slot.
 ///
-/// For some VMs, a slot may sometimes not hold an object reference.  For example, it can hold a
-/// special `NULL` pointer which does not point to any object, or it can hold a tagged
-/// non-reference value, such as small integers and special values such as `true`, `false`, `null`
-/// (a.k.a. "none", "nil", etc. for other VMs), `undefined`, etc.
+/// A slot can be in an object (a.k.a. a field), on the stack (i.e. a local variable) or any other
+/// places (such as global variables).  A slot may hold an object reference, but for some VMs, a
+/// slot may sometimes not hold an object reference.  For example, in JVM, a slot can hold a special
+/// `null` pointer which does not point to any object; in some VMs, including CRuby and some
+/// JavaScript engines, a slot can hold a tagged non-reference value, such as small integers and
+/// special values such as `true`, `false`, `null` (a.k.a. "none", "nil", etc.), `undefined`, etc.
 ///
-/// This intends to abstract out the differences of reference field representation among different
-/// VMs.  If the VM represent a reference field as a word that holds the pointer to the object, it
-/// can use the default `SimpleSlot` we provide.  In some cases, the VM need to implement its own
-/// `Slot` instances.
+/// The `Slot` trait is intended to abstract out the differences of reference field representation
```

Comment:
> The `Slot` trait is intended to abstract out the differences of reference field representation
> (compressed, tagged, offsetted, etc.) among different VMs.

I feel this is also a good definition for `Slot`. It is just an abstraction. It does not really define what a `Slot` would be -- whether it is a value that points to the slot, or is the slot itself. It doesn't really matter, as long as it allows MMTk to access the object reference in the memory.